src/Symfony/Component/Serializer/Encoder/YamlEncoder.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Encoder;
+
+use Symfony\Component\Yaml\Dumper;
+use Symfony\Component\Yaml\Parser;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Encodes YAML data.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class YamlEncoder implements EncoderInterface, DecoderInterface
+{
+    const FORMAT = 'yaml';
+
+    private $dumper;
+    private $parser;
+    private $defaultContext;
+
+    public function __construct(Dumper $dumper = null, Parser $parser = null, $defaultContext = array('yaml_inline' => 0, 'yaml_indent' => 0, 'yaml_flags' => 0))
```

Comment:
rather than putting a default value here, you should merge the default with the argument being received.
Otherwise, you have no guarantee about the available keys (and so your access to context options will be unsafe)